diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..9d24757
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,63 @@
+# Not caring about other platforms, only Linux via X11.
+
+cc_library(
+    name = "glfw_src",
+    srcs = glob([
+        "src/context.c",
+        "src/init.c",
+        "src/input.c",
+        "src/monitor.c",
+        "src/platform.c",
+        "src/vulkan.c",
+        "src/window.c",
+        "src/egl_context.c",
+        "src/osmesa_context.c",
+        "src/null_init.c",
+        "src/null_monitor.c",
+        "src/null_window.c",
+        "src/null_joystick.c",
+        "src/posix_module.c",
+        "src/posix_time.c",
+        "src/posix_thread.c",
+        "src/x11_init.c",
+        "src/x11_monitor.c",
+        "src/x11_window.c",
+        "src/xkb_unicode.c",
+        "src/glx_context.c",
+        "src/linux_joystick.c",
+        "src/posix_poll.c",
+    ]),
+    hdrs = glob([
+        "src/internal.h",
+        "src/platform.h",
+        "src/mappings.h",
+        "src/null_platform.h",
+        "src/null_joystick.h",
+        "src/posix_time.h",
+        "src/posix_thread.h",
+        "src/x11_platform.h",
+        "src/xkb_unicode.h",
+        "src/linux_joystick.h",
+        "src/posix_poll.h",
+        "include/GLFW/glfw3.h",
+        "include/GLFW/glfw3native.h",
+    ]),
+    defines = [
+        "_GLFW_HAS_XF86VM",
+        "_GLFW_X11",
+    ],
+    linkopts = [
+        "-lX11",
+    ]
+)
+
+cc_library(
+    name = "glfw",
+    hdrs = glob([
+        "include/GLFW/glfw3.h",
+        "include/GLFW/glfw3native.h",
+    ]),
+    deps = [":glfw_src"],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
+)

diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..d30394d
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,6 @@
+
+module(
+    name = "glfw",
+    version = "3.4.0.tfr.1"
+)
+
